packages/data-point/lib/entity-types/base-entity/resolve.js

```diff
@@ -167,6 +171,8 @@ function resolveEntity (
         error,
         currentAccumulator,
         resolveReducer
+      ).then(acc =>
+        typeCheck(manager, acc, acc.reducer.spec.outputType, resolveReducer)
```

Comment:
@raingerber This is now executing `typeCheck` against the `error` method's return value. We are not catching errors thrown from the `error` method